src/document/doc-builders.js

```diff
@@ -257,4 +259,6 @@ module.exports = {
   markAsRoot,
   dedentToRoot,
   dedent,
+  hardlineNoBreak,
```

Comment:
These is going public, better add comment why they are added, so we can remove if we don't need anymore.

Another thing I'm not sure if you know, this is actually used in codebase https://github.com/prettier/prettier/blob/1df287acc802939edfd5eed3ffabf3979a8cc3ae/src/language-markdown/printer-markdown.js#L560,  did you consider this when you make changes for the debug-print

Reply:
The fact that they're used is exactly the reason why we need to be able to print them in debug-print.

Reply:
> we can remove if we don't need anymore

Why would that happen?

Reply:
Have a look at `show doc` [here](https://deploy-preview-10169--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAFAlgGwjAPgQXQCcAHCIvACQgFciBnAHSgFoXck2lcvmApCAAsoAAgAiEOLgDKAaQDiuAIwBmFQA5+AQyhwRAJUm4+AMVm4VSgCzMQAGhAQSMdNHrJQWokQgB3VF4I7ihamL5aAJ7uDgBGRFpgANZwMNIkCehQAObIMEQ0cA5wALYxcAAm5RUAMjpZNFpZcCbkxVowLtnIIFo0MBD2IIIwxZgA6oLo8PTpYHDSQVPoAG5TEd1g9NEgmfRwFKjxWW3IAGahew4AVvQAHgBC8Ukp0lrFcNWZcGcXhSA3t2kmSymDgAEUaDhvkhzphLiB0gx9t02kREuU-FBBiQiJkYGN0OUYIJkOoAAwOHEQPZjeIkbo4uB7IjLb4OACOkPghycwR69BYugqFUGRDgnOIcEOjROMN+Dj2xXQuXyf3owNBEKhPzhfxgWhiBKJJKQACYHHktFhgQBhCDFWUgJkAVkGND2ABUDcFYfDlgUAJJQKqwaRgXHOfDB6QwCKgnV7AC+iaAA).

Reply:
> Why would that happen?

Point is, adding things are  easy, but  hard 
to  remove things , it's not documented. Have you ever imagine we are removing `concat()`? Things change. All I ask it add a  comment mention it's not intend to let people use.

Reply:
It's used already for Markdown and at least in the Ruby plugin. So I'd rather document it and let people use it directly, not via `getDocParts`.

Reply:
Fine to me.